cwms-data-api/src/main/java/cwms/cda/data/dao/project/ProjectChildrenDao.java

```diff
@@ -0,0 +1,262 @@
+
+        List<String> allKinds = getProjectKinds();
+
+        Set<String> kinds = getMatchingKinds(kindRegex, allKinds);
+
+        return children(office, projLike, kinds);
+    }
+
+    private List<ProjectChildren> children(String office, String projLike, Set<String> kinds) {
+
+        Map<String, ProjectChildren.Builder> builderMap = new LinkedHashMap<>();  // proj-id->
+
+        for (String kind : kinds) {
+            Map<String, List<CwmsId>> locsOfKind = getChildrenOfKind(office, projLike, kind);
+            if (locsOfKind != null) {
+                for (Map.Entry<String, List<CwmsId>> entry : locsOfKind.entrySet()) {
+                    String projId = entry.getKey();
+                    List<CwmsId> locs = entry.getValue();
+                    ProjectChildren.Builder builder = builderMap.computeIfAbsent(projId, k ->
+                            new ProjectChildren.Builder()
+                                    .withProject(new CwmsId.Builder()
+                                            .withOfficeId(office)
+                                            .withName(projId)
+                                            .build()));
+                    switch (kind) {
+                        case "EMBANKMENT":
+                            builder.withEmbankments(locs);
+                            break;
+                        case "LOCK":
+                            builder.withLocks(locs);
+                            break;
+                        case "OUTLET":
+                            builder.withOutlets(locs);
+                            break;
+                        case "TURBINE":
+                            builder.withTurbines(locs);
+                            break;
+                        case "GATE":
+                            builder.withGates(locs);
+                            break;
+                        default:
+                            break;
+                    }
+
+                }
+            }
+        }
+
+        return builderMap.values().stream()
+                .map(ProjectChildren.Builder::build)
+                .collect(Collectors.toList());
+
+    }
+
+    @Nullable
+    private Map<String, List<CwmsId>> getChildrenOfKind(String office, String projLike, String kind) {
+        switch (kind) {
+
+            case "EMBANKMENT":
+                return getEmbankmentChildren(office, projLike);
+            case "TURBINE":
+                return getTurbineChildren(office, projLike);
+            case "OUTLET":
+                return getOutletChildren(office, projLike);
+            case "LOCK":
+                return getLockChildren(office, projLike);
+            case "GATE":
+                return getGateChildren(office, projLike);
+            default:
+                return null;
+        }
+
+    }
+
+    private Map<String, List<CwmsId>> getGateChildren(String office, @Nullable String projLike) {
```

Comment:
You could code this Map as <CwmsId, List<CwmsId>> to provide an office ID to the project.

Reply:
Wouldn't that require CwmsId to have working implementations of hashCode and equals?

Reply:
Right.  I guess I don't understand @rma-psmorris comment.   In the discussion of removing hashCode and equals from all the DTO type I explained how they could be used in Dao implementations.  My example was LocationGroup and LocationCategory but the same can be said for CwmsId.  Why would CwmsId as a key be better than projId?  Do you want that so that we can make the office parameter optional?  Do any of the project-child allow office mismatch between the project and the project-child locations?

Reply:
Yeah, you are both correct. It would have to be a concatenated String, not a CwmsId. 

The intent of my suggestion was to bubble up the office/project in the return rather than just the project. Up above in the controller you are connecting the queried project id to the argument office so the returned JSON is complete.